### PR TITLE
Add missing wrapper

### LINKED ISSUE #2686: Missing Wrapper
### Behavior

When Bundler is installed via a `gem update --system`, the wrapper files normally generated by `gem install` are missing. Notably the `bundle` command is the `exe/bundle` file from the Bundler gem.

### Reproduction

Start image:

```
$ docker run -it --rm ruby:2.5 bash
```

```
root@911c503c3cc4:/# which bundle
/usr/local/bin/bundle
root@911c503c3cc4:/# bundle _1.17.3_ version
Could not find command "_1.17.3_".
root@911c503c3cc4:/# gem install bundler --version 2.0.1
Fetching bundler-2.0.1.gem
Successfully installed bundler-2.0.1
1 gem installed
root@911c503c3cc4:/# hash -r
root@911c503c3cc4:/# which bundle
/usr/local/bundle/bin/bundle
root@911c503c3cc4:/# bundle _1.17.3_ version
Bundler version 1.17.3 (2019-03-04 commit d7089abb6)
root@911c503c3cc4:/#
```

Note that the `hash -r` is required otherwise this will happen:

```
root@e86383a3c9fa:/# which bundle
/usr/local/bin/bundle
root@e86383a3c9fa:/# type bundle
bundle is /usr/local/bin/bundle
root@e86383a3c9fa:/# bundle _1.17.3_ version
Could not find command "_1.17.3_".
root@e86383a3c9fa:/# type bundle
bundle is hashed (/usr/local/bin/bundle)
root@e86383a3c9fa:/# gem install bundler --version 2.0.1
Fetching bundler-2.0.1.gem
Successfully installed bundler-2.0.1
1 gem installed
root@e86383a3c9fa:/# which bundle
/usr/local/bundle/bin/bundle
root@e86383a3c9fa:/# type bundle
bundle is hashed (/usr/local/bin/bundle)
root@e86383a3c9fa:/# bundle _1.17.3_ version
Could not find command "_1.17.3_".
```

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
# Description:

Fixes #2686.

When the default bundler gem was being installed through `gem update --system`, the proper rubygems wrapper was not being generated. This PR fixes that, and also adds missing wrapper support for `gem install --default` in general.

# Tasks:

- [x] Describe the problem / feature
- [x] Write tests
- [x] Write code to solve the problem
- [ ] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: lib/rubygems/commands/setup_command.rb [hunk 1]
```diff
@@ -427,13 +426,12 @@ def install_default_bundler_gem
...
+    require 'rubygems/installer'
 
-      installer = Gem::Installer.for_spec bundler_spec
-      bundler_spec.executables.each do |e|
-        installer.generate_windows_script e, bundler_spec.bin_dir
-      end
+    Dir.chdir("bundler") do
+      built_gem = Gem::Package.build(bundler_spec)
+      installer = Gem::Installer.at(built_gem, env_shebang: options[:env_shebang], install_as_default: true, bin_dir: bin_dir, wrappers: true)
+      installer.install
```
Comment (deivid-rodriguez) [inline lib/rubygems/commands/setup_command.rb]: I should cleanup after myself here.

**Comment by hsbt** [comment]
Thanks, I forgot to update this behavior same as Ruby 2.6.

**Comment by hsbt** [comment]
@bundlerbot r+

**Comment by deivid-rodriguez** [comment]
@hsbt Are you planning to backport this fix to 3.0?

**Pull request merged**